File: pocketr-java.api/src/main/java/com/decrux/pocketr/api/entities/dtos/HouseholdMemberDto.java
package com.decrux.pocketr.api.entities.dtos;

import java.time.Instant;

public record HouseholdMemberDto(
    long userId,
    String email,
    String firstName,
    String lastName,
    String role,
    String status,
    Instant joinedAt
) {
    public long getUserId() {
        return userId;
    }

    public String getEmail() {
        return email;
    }

    public String getFirstName() {
        return firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public String getRole() {
        return role;
    }

    public String getStatus() {
        return status;
    }

    public Instant getJoinedAt() {
        return joinedAt;
    }
}

File: pocketr-api/src/main/kotlin/com/decrux/pocketr_api/entities/dtos/HouseholdMemberDto.kt
package com.decrux.pocketr_api.entities.dtos

import java.time.Instant

data class HouseholdMemberDto(
    val userId: Long,
    val email: String,
    val firstName: String?,
    val lastName: String?,
    val role: String,
    val status: String,
    val joinedAt: Instant?,
)
